### PR TITLE
feat: new nft badge

### LINKED ISSUE #4614: [Bug] Failed to inject NFT avatar box into the profile dialog
### How to reproduce?

1. Goto profile page https://twitter.com/[twitter_handle]
2. Open the settings dialog and change Twitter default font size 
3. Click the NFT avatar button to the profile dialog

### Mask Version:

- [ ] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [ ] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [x] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [x] Latest Chrome (stable channel)
- [ ] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by lelenei**
closes #4614

### CODE REVIEW

File: packages/maskbook/src/social-network-adaptor/twitter.com/ui-provider.ts [hunk 1]
```diff
@@ -171,6 +173,8 @@ const twitterUI: SocialNetworkUI.Definition = {
         enhancedProfileNFTAvatar: injectProfileNFTAvatarInTwitter,
         profileAvatar: injectNFTAvatarInTwitter,
         openNFTAvatar: injectOpenNFTAvatarEditProfileButton,
+        tweetNFTAvatar: injectUserNFTAvatarAtTweet,
+        CollectionAvatar: injectNFTContractAtTwitter,
```
Comment (guanbinrui) [inline packages/maskbook/src/social-network-adaptor/twitter.com/ui-provider.ts:177]:
```suggestion
        collectionAvatar: injectNFTContractAtTwitter,
```

To keep the same code style with others.

File: packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarRing.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            <svg className={classes.root} width={R} height={R} viewBox={`0 0 ${R} ${R}`} id="NFTAvatarRingIcon">
+                <defs>
+                    <path
+                        id="path"
+                        fill="none"
+                        stroke="none"
+                        strokeWidth="0"
+                        d={`M${x1} ${y1} A${path_r} ${path_r} 0 1 1 ${x2} ${y2}`}
+                    />
+                    <linearGradient id="gradient" x1="0%" y1="0%" x2="100%" y2="0">
```
Comment (UncleBill) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarRing.tsx:40]: Create an uniq id for every rendering

```suggestion
                    <linearGradient id={uniqId} x1="0%" y1="0%" x2="100%" y2="0">
```

File: packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarRing.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    const r = R / 2 - strokeWidth
+    const path_r = R / 2 - strokeWidth + fontSize / 2
+    const x1 = R / 2 - path_r / 2
+    const y1 = R / 2 + Math.sqrt(Math.pow(path_r, 2) - Math.pow(path_r / 2, 2))
+    const x2 = R / 2 + path_r / 2
+    const y2 = y1
+
+    return (
+        <RainbowBox width={R} height={R}>
+            <svg className={classes.root} width={R} height={R} viewBox={`0 0 ${R} ${R}`} id="NFTAvatarRingIcon">
```
Comment (UncleBill) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarRing.tsx:31]:
```suggestion
            <svg className={classes.root} width={R} height={R} viewBox={`0 0 ${R} ${R}`}>
```

File: packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarRing.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+                <circle
+                    cx={R / 2}
+                    cy={R / 2}
+                    r={r + strokeWidth / 2}
+                    fill="none"
+                    stroke={stroke}
+                    strokeWidth={strokeWidth}
+                />
+                <pattern id="pattern" x="0" y="0" width="300%" height="100%" patternUnits="userSpaceOnUse">
```
Comment (UncleBill) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarRing.tsx:58]: ditto
```suggestion
                <pattern id={uniqPatterId} x="0" y="0" width="300%" height="100%" patternUnits="userSpaceOnUse">
```

File: packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/profileNFTAvatar.tsx [hunk 5]
```diff
@@ -73,7 +73,7 @@ function NFTAvatarInTwitter() {
         return () => profileSave.removeEventListener('click', handler)
     }, [handler])
 
-    if (myIdentities.some((x) => x && x.userId === identity.identifier.userId))
-        return <NFTAvatar onChange={onChange} classes={classes} />
-    return null
+    //if (myIdentities.some((x) => x && x.userId === identity.identifier.userId))
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/injection/NFT/profileNFTAvatar.tsx:76]: Remove it.

**Comment by guanbinrui** [comment]
@lelenei, Would you please open a new PR for fixing the OpenSea related bug?

**Pull request merged**